app/src/db/migrations/20210721210234_019-api-key-models.js

```diff
@@ -0,0 +1,170 @@
+      const items = [
+        {
+          createdBy: CREATED_BY,
+          code: 'form_api_create',
+          display: 'API Key Create',
+          description: 'Can create an api key for a form',
+          active: true
+        },
+        {
+          createdBy: CREATED_BY,
+          code: 'form_api_read',
+          display: 'API Key Read',
+          description: 'Can view the API key for a form (unencrypted)',
+          active: true
+        },
+        {
+          createdBy: CREATED_BY,
+          code: 'form_api_update',
+          display: 'API Key Update',
+          description: 'Can update the API key for a form',
+          active: true
+        },
+        {
+          createdBy: CREATED_BY,
+          code: 'form_api_delete',
+          display: 'API Key Delete',
+          description: 'Can remove the API key from a form',
+          active: true
+        }
+      ];
+      return knex('permission').insert(items).returning('code');
+    })
+
+    // Form Owners (and only owners now) can do these new permissions
+    .then(() => {
+      const rolePermssion = {
+        id: uuidv4(),
+        createdBy: CREATED_BY,
+        role: 'owner',
+        permission: 'form_api_create'
+      };
+      return knex('role_permission').insert(rolePermssion);
+    })
+
+    .then(() => {
+      const rolePermssion = {
+        id: uuidv4(),
+        createdBy: CREATED_BY,
+        role: 'owner',
+        permission: 'form_api_read'
+      };
+      return knex('role_permission').insert(rolePermssion);
+    })
+
+    .then(() => {
+      const rolePermssion = {
+        id: uuidv4(),
+        createdBy: CREATED_BY,
+        role: 'owner',
+        permission: 'form_api_update'
+      };
+      return knex('role_permission').insert(rolePermssion);
+    })
+
+    .then(() => {
+      const rolePermssion = {
+        id: uuidv4(),
+        createdBy: CREATED_BY,
+        role: 'owner',
+        permission: 'form_api_delete'
+      };
+      return knex('role_permission').insert(rolePermssion);
+    })
+
+    // Add the new table for API keys
+    .then(() => knex.schema.createTable('form_api_key', table => {
+      table.specificType(
+        'id',
+        'integer GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY'
+      );
+      table.uuid('formId').references('id').inTable('form').notNullable().index();
+      table.uuid('secret').unique().notNullable();
+      stamps(knex, table);
+    }))
+
+    // Add the new table for API key audit history
+    .then(() => knex.schema.createTable('form_api_key_audit', table => {
+      table.specificType(
+        'id',
+        'integer GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY'
+      );
+      table.integer('keyId').notNullable();
+      table.uuid('formId').notNullable().index();
+      table.string('dbUser').notNullable();
+      table.string('updatedByUsername');
+      table.timestamp('actionTimestamp', { useTz: true }).defaultTo(knex.fn.now());
+      table.string('action').notNullable();
```

Comment:
Wonder if this should be an enum instead of a string?
https://knexjs.org/#Schema-enum